snuba/web/rpc/storage_routing/routing_strategies/storage_routing.py

```diff
@@ -348,31 +460,65 @@ def get_routing_decision(self, routing_context: RoutingContext) -> RoutingDecisi
             return routing_decision
 
     @final
-    def output_metrics(self, routing_decision: RoutingDecision) -> None:
-        assert routing_decision.routing_context is not None
-        self._emit_routing_mistake(routing_decision)
-        self._output_metrics(routing_decision.routing_context)
-        query_result = routing_decision.routing_context.query_result or QueryResult(
-            {}, {"stats": {}, "sql": "", "experiments": {}}
-        )
-        profile = query_result.result.get("profile", {}) or {}
-        if elapsed := profile.get("elapsed"):
-            self._record_value_in_span_and_DD(
-                routing_context=routing_decision.routing_context,
-                metrics_backend_func=self.metrics.timing,
-                name="query_timing",
-                value=elapsed,
-                tags={"tier": routing_decision.tier.name},
+    def after_execute(self, routing_decision: RoutingDecision, error: Exception | None) -> None:
+        try:
+            assert routing_decision.routing_context is not None
+
+            routing_decision.strategy.update_allocation_policies_balances(routing_decision, error)
```

Comment:
shouldn't this just be calling `self._get_combined_allocation_policies_recommendations` ? There's unnecessary indirection here